src/StaticWebAssetsSdk/Tasks/DefineStaticWebAssets.Cache.cs

```diff
@@ -0,0 +1,262 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+using Microsoft.AspNetCore.StaticWebAssets.Tasks.Utils;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using static Microsoft.AspNetCore.StaticWebAssets.Tasks.FingerprintPatternMatcher;
+
+namespace Microsoft.AspNetCore.StaticWebAssets.Tasks;
+
+public partial class DefineStaticWebAssets : Task
+{
+    private DefineStaticWebAssetsCache GetOrCreateAssetsCache()
+    {
+        var assetsCache = DefineStaticWebAssetsCache.ReadOrCreateCache(Log, CacheManifestPath);
+        if (CacheManifestPath == null)
+        {
+            assetsCache.NoCache(CandidateAssets);
+            return assetsCache;
+        }
+
+        var memoryStream = new MemoryStream();
+#if NET9_0_OR_GREATER
+        Span<string> properties = new[] {
```

Comment:
Unfortunately using `new[]` will allocate an array unconditionally: https://sharplab.io/#v2:CYLg1APgAgTAjAWAFAGUAOBDAdgHinABgH4A+AAgEsAXAUwFsBnMgXjKxoHcBtAXTIG8yAIjhCANMJjjhAZmlCALPICs8gGzyA7ELIBfANzJ8ATgAU+AgDoAUgHsKWUwHIxLyrUYBKT/qA==.

I think the optimization only applies if you use a collection expression (`[]`). Feel free to not waste time on this if it's too much a hassle to do in a way that works across .NET versions 😄